dd-library-php-setup.php

```
@@ -0,0 +1,833 @@
+        if (is_alpine()) {
+            $extensionSuffix = '-alpine';
+        } elseif (is_truthy($phpProperties[IS_DEBUG])) {
+            $extensionSuffix = '-debug';
+        } elseif (is_truthy(THREAD_SAFETY)) {
+            $extensionSuffix = '-zts';
+        }
+        $extensionRealPath = $tmpExtensionsDir . '/ddtrace-' . $extensionVersion . $extensionSuffix . '.so';
+        $extensionFileName = 'ddtrace.so';
+        $extensionDestination = $phpProperties[EXTENSION_DIR] . '/' . $extensionFileName;
+
+        // Move - rename() - instead of copy() since copying does a fopen() and copies to the stream itself, causing a
+        // segfault in the PHP process that is running and had loaded the old shared object file.
+        $tmpExtName = $extensionDestination . '.tmp';
+        copy($extensionRealPath, $tmpExtName);
+        rename($tmpExtName, $extensionDestination);
+        echo "Copied '$extensionRealPath' '$extensionDestination'\n";
+
+        // Writing the ini file
+        $iniFileName = '98-ddtrace.ini';
+        $iniFilePaths = [$phpProperties[INI_CONF] . '/' . $iniFileName];
+        if (\strpos('/cli/conf.d', $phpProperties[INI_CONF]) >= 0) {
+            // debian based distros have INI folders split by SAPI, in a predefined way:
+            //   - <...>/cli/conf.d       <-- we know this from php -i
+            //   - <...>/apache2/conf.d   <-- we derive this from relative path
+            //   - <...>/fpm/conf.d       <-- we derive this from relative path
+            $apacheConfd = str_replace('/cli/conf.d', '/apache2/conf.d', $phpProperties[INI_CONF]);
+            if (\is_dir($apacheConfd)) {
+                array_push($iniFilePaths, "$apacheConfd/$iniFileName");
+            }
+        }
+        foreach ($iniFilePaths as $iniFilePath) {
+            if (!file_exists($iniFilePath)) {
+                file_put_contents($iniFilePath, get_ini_template($installDirWrapperPath));
+                echo "Created INI file '$iniFilePath'\n";
```

Handle the error case, e.g. write permission failure instead of reporting success after the warning from PHP.